Check whether User-Agent is defined before trying to override it

Since Go 1.20 [1], ReverseProxy no longer adds a User-Agent header to forwarded
requests.

[1] https://github.com/golang/go/commit/f001df540b3fc66a475985c1b7c810e7df063c8f

--- a/reverseproxy/reverseproxy.go
+++ b/reverseproxy/reverseproxy.go
@@ -32,7 +32,7 @@ func newDirector(target *url.URL, extraDirector func(*http.Request)) func(*http.
 		} else {
 			req.URL.RawQuery = targetQuery + "&" + req.URL.RawQuery
 		}
-		if _, ok := req.Header["User-Agent"]; !ok {
+		if req.Header.Get("User-Agent") != "" {
 			// explicitly disable User-Agent so it's not set to default value
 			req.Header.Set("User-Agent", "")
 		}
